### ISSUE TITLE
Two planets with same SMA cause sentinel telescope code to cause exceptions

### ISSUE THREAD

**Issue by StormCircuit** [Kopernicus/Kopernicus#605]
Over at KSPCF an issue was found with stock ksp in which any two celestial bodies with the same SMA will cause exceptions. It is explained on the KSPCF issue tracker in detail here:
https://github.com/KSPModdingLibs/KSPCommunityFixes/issues/158

I am uncertain if this issue even belongs here but considering Kopernicus is a mod for creating planets and specific combinations of planets can cause this (without checks as far as I know) it seemed pertinent.

**Comment by JonnyOThan** [comment]
I originally filed that bug under KSPCF because it's a bug in the stock code - but since it only affects modded systems and people might not have KSPCF installed, AND kopernicus now depends on harmony anyway then it makes more sense to fix it here.

**Comment by R-T-B** [comment]
Yeah, this is a big that fits perfectly into Kopernicus's scope.  I'll see about it for next release, provided @gotmachine agrees.  Pinning this so as to remember it.

**Comment by R-T-B** [comment]
As some potential field research in the interim if anyone is curious (next release won't be for a few weeks at least):

Does this only occur when the asteroid generator is set to "stock?" OR does it occur with "true" and "false" settings as well?

Thanks.

**Comment by gotmachine** [comment]
This has nothing to do with asteroid generation. The problematic code is used by the stock sentinel partmodule and the sentinel contracts, it doesn't care what was used to generate asteroids.

And yes, as mentioned in the KSPCF issue, I think this should be fixed by Kopernicus, as while this is a stock bug, it can only happen in the presence of a modified planetary system.

**Comment by R-T-B** [comment]
> This has nothing to do with asteroid generation. The problematic code is used by the stock sentinel partmodule and the sentinel contracts, it doesn't care what was used to generate asteroids.
> 
> And yes, as mentioned in the KSPCF issue, I think this should be fixed by Kopernicus, as while this is a stock bug, it can only happen in the presence of a modified planetary system.

I'm aware got.  When Kopernicus changes the generator, it also completely obliterates the stock module for sentinel contract management, IIRC.  We do that to prevent the contracts from spawning and being broken.  Hence me comment.
